// ngx-pretty-checkbox/esm2015/lib/components/ngx-pretty-toggle-will-change/ngx-pretty-toggle-will-change.component.js
import { Input, ElementRef, Component, Renderer2, ChangeDetectionStrategy } from '@angular/core';
import { PrettyCheckBoxToggleType } from '../../model/interfaces';
import { getColorClassName } from '../../utility';
export class NgxPrettyToggleWillChangeComponent {
    constructor(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.outline = false;
    }
    get isToggleOn() { return this.type === PrettyCheckBoxToggleType.On; }
    get isToggleOff() { return this.type === PrettyCheckBoxToggleType.Off; }
    ngOnChanges(changes) {
        if (changes.color || changes.outline) {
            const oldClass = getColorClassName(changes.color ? changes.color.previousValue : this.color, changes.outline ? changes.outline.previousValue : null);
            const newClass = getColorClassName(changes.color ? changes.color.currentValue : this.color, changes.outline ? changes.outline.currentValue : null);
            this.renderer.removeClass(this.el.nativeElement, oldClass);
            this.renderer.addClass(this.el.nativeElement, newClass);
        }
    }
}
NgxPrettyToggleWillChangeComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-p-toggle[will-change], p-toggle[will-change]',
                template: "<ng-content select=\"[pIcon], [p-icon], [pSvg], [p-svg], [pImage], [p-image]\"></ng-content>\r\n<label><ng-content></ng-content></label>",
                host: {
                    'class': 'state',
                    '[class.p-on]': 'isToggleOn',
                    '[class.p-off]': 'isToggleOff',
                },
                changeDetection: ChangeDetectionStrategy.OnPush
            },] }
];
NgxPrettyToggleWillChangeComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NgxPrettyToggleWillChangeComponent.propDecorators = {
    type: [{ type: Input }],
    color: [{ type: Input }],
    outline: [{ type: Input }]
};
