// ngx-pretty-checkbox/esm2015/lib/model/interfaces.js
export var PrettyCheckBoxToggleType;
(function (PrettyCheckBoxToggleType) {
    PrettyCheckBoxToggleType["On"] = "on";
    PrettyCheckBoxToggleType["Off"] = "off";
})(PrettyCheckBoxToggleType || (PrettyCheckBoxToggleType = {}));
export var PrettyCheckboxStroke;
(function (PrettyCheckboxStroke) {
    // Accept Null
    PrettyCheckboxStroke["Fill"] = "fill";
    PrettyCheckboxStroke["Thick"] = "thick";
    PrettyCheckboxStroke["Slim"] = "slim"; // Switch
})(PrettyCheckboxStroke || (PrettyCheckboxStroke = {}));
export var PrettyCheckboxShape;
(function (PrettyCheckboxShape) {
    // Accept Null
    PrettyCheckboxShape["Curve"] = "curve";
    PrettyCheckboxShape["Round"] = "round";
})(PrettyCheckboxShape || (PrettyCheckboxShape = {}));
export var PrettyCheckboxColor;
(function (PrettyCheckboxColor) {
    PrettyCheckboxColor["Primary"] = "primary";
    PrettyCheckboxColor["Success"] = "success";
    PrettyCheckboxColor["Info"] = "info";
    PrettyCheckboxColor["Warning"] = "warning";
    PrettyCheckboxColor["Danger"] = "danger";
})(PrettyCheckboxColor || (PrettyCheckboxColor = {}));
export var PrettyCheckboxAnimation;
(function (PrettyCheckboxAnimation) {
    PrettyCheckboxAnimation["Smooth"] = "smooth";
    PrettyCheckboxAnimation["Jelly"] = "jelly";
    PrettyCheckboxAnimation["Tada"] = "tada";
    PrettyCheckboxAnimation["Rotate"] = "rotate";
    PrettyCheckboxAnimation["Pulse"] = "pulse";
})(PrettyCheckboxAnimation || (PrettyCheckboxAnimation = {}));
